### PR TITLE
Track SCIM users + API response success/error counts

### PR DESCRIPTION
**Pull request by noahmoss**
* Tracks SCIM-created users under the `:sso_source` key and includes a count in the stats ping of the number of SCIM users created in the last 24 hours
  * Using `sso_source` for this isn't totally accurate since it's not strictly an SSO method, but I think this is the most logical place to track this information
* Adds Prometheus counters for SCIM success and error responses, and adds a macro to wrap each SCIM endpoint and increment the counters when appropriate

### CODE REVIEW

**Comment by noahmoss** [comment]
* **#48702** <a href="https://app.graphite.dev/github/pr/metabase/metabase/48702?utm_source=stack-comment-icon" target="_blank">Graphite</a> 👈
* `master`

This stack of pull requests is managed by Graphite. <a href="https://stacking.dev/?utm_source=stack-comment">Learn more about stacking.</a>
<h2></h2>

 Join @noahmoss and the rest of your teammates on <a href="https://graphite.dev?utm-source=stack-comment">Graphite <b>Graphite</b></a>

File: enterprise/backend/src/metabase_enterprise/scim/v2/api.clj [hunk 1]
```diff
@@ -188,98 +206,103 @@
...
-        result         {:schemas      [list-schema-uri]
-                        :totalResults (t2/count :model/User {:where where-clause})
-                        :startIndex   (inc offset)
-                        :itemsPerPage items-per-page
-                        :Resources    (map mb-user->scim hydrated-users)}]
-    (scim-response result)))
+  (with-prometheus-counters
+    (let [limit          (or c default-pagination-limit)
+          ;; SCIM start-index is 1-indexed, so we need to decrement it here
+          offset         (if start-index (dec start-index) default-pagination-offset)
```
Comment (escherize) [inline enterprise/backend/src/metabase_enterprise/scim/v2/api.clj:212]: Will limit be off-by-one if start-index gets dec'd?

File: enterprise/backend/test/metabase_enterprise/scim/v2/api_test.clj [hunk 2]
```diff
@@ -142,8 +165,9 @@
                           :active true}
                 response (scim-client :post 201 "ee/scim/v2/Users" new-user)]
             (is (malli= scim-api/SCIMUser response))
-            (is (=? (select-keys user [:email :first_name :last_name :is_active])
-                    (t2/select-one [:model/User :email :first_name :last_name :is_active]
+            (is (=? (assoc (select-keys user [:email :first_name :last_name :is_active])
```
Comment (escherize) [inline enterprise/backend/test/metabase_enterprise/scim/v2/api_test.clj:168]: nit: Could this be `=` instead of `=?`?

**Review by escherize — APPROVED** [review]
LGTM

File: enterprise/backend/src/metabase_enterprise/scim/v2/api.clj (hunk 1 above)
Comment (noahmoss) [reply enterprise/backend/src/metabase_enterprise/scim/v2/api.clj:212]: I don't think so? limit just controls the number of values fetched at once and will be 100 (default value) for most calls. So the client will send `start-index=1`, `start-index=101`, etc, and we'll fetch the same users using offsets of 0, 100, etc.

File: enterprise/backend/test/metabase_enterprise/scim/v2/api_test.clj (hunk 2 above)
Comment (noahmoss) [reply enterprise/backend/test/metabase_enterprise/scim/v2/api_test.clj:168]: it's subtle, but fetching a user also hydrates the `common_name` field automatically, so it's not exactly equal. a little annoying

**Pull request merged**